tests/db/model/collector/methods.js

```diff
@@ -288,12 +285,12 @@ describe('tests/db/model/collector/methods.js >', () => {
         .then((c1) => c1.reassignGenerators())
 
         .then(() => Promise.join(
-          Generator.find({ where: { name: generator1.name } }),
-          Generator.find({ where: { name: generator2.name } }),
-          Generator.find({ where: { name: generator3.name } }),
+          Generator.findOne({ where: { name: generator1.name } }),
+          Generator.findOne({ where: { name: generator2.name } }),
+          Generator.findOne({ where: { name: generator3.name } }),
         ))
         .spread((gen1, gen2, gen3) => {
-          expect(gen1.currentCollector.name).to.equal(collector2.name);
+          expect(gen1.currentCollector.name).to.not.equal(collector1.name);
```

Comment:
So the problem was that gens are being assigned inconsistently?
This could have been triggered by the change to balancing the assignment (these tests were written before that).  But I think that should still be deterministic. 
It could be that the test setup is creating things in parallel, and the order those operations execute in can determine where things get assigned down the line.

Reply:
thx i'll look into that some more